### PR TITLE
Update "Followers" terminology to "Subscribers"

### LINKED ISSUE #20690: Update terminology to Subscribers from Followers
Full Scope: https://github.com/Automattic/wordpress-mobile/issues/47
Design: [IqhXWz3Iir7RMb5XH5gGfZ-fi-883%3A4006](https://href.li/?https://www.figma.com/file/IqhXWz3Iir7RMb5XH5gGfZ/?node-id=883%3A4006)

Update terminology to Subscribers from Followers in all the app

### PR DESCRIPTION
**Pull request by aditi-bhatia**
Fixes #20690 
This PR updates `followers` and `follows` terminology in the app to `subscribers`. Text related to following tags or conversations should not be affected, as well as any other usage of `follow` that doesn't fit within the `subscribers` context.

-----

## To Test:
1. To test, check `Stats`, `Notifications`, `Reader`, and `People` since this is where most of the changes occur. 

Extra: Review strings.xml for usages of `followers`, `follows`, `follow`, and `following` to ensure all cases have been updated.

Some examples:

|                                | Before | After |
|--------------------------------|--------|-------|
| Stats -> Insights tab -> No Subscribers |  Screenshot 2024-05-06 at 6 40 37 PM    | Screenshot 2024-05-06 at 4 26 34 PM   |
| Stats -> Insights tab -> Subscribers    |  Screenshot 2024-05-06 at 6 42 37 PM      |    Screenshot 2024-05-06 at 6 02 12 PM |
| Notifications -> Follows tab   |   Screenshot 2024-05-06 at 6 44 28 PM     |    Screenshot 2024-05-06 at 5 53 42 PM  |
| My Site -> More -> People -> `+` button  |      Screenshot 2024-05-06 at 6 31 59 PM  |  Screenshot 2024-05-06 at 6 32 09 PM    |
| My Site -> More -> People -> `Team` dropdown menu  |    Screenshot 2024-05-06 at 6 34 45 PM    |    Screenshot 2024-05-06 at 6 34 35 PM  |

-----

## Regression Notes

1. Potential unintended areas of impact

    - String translations

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

4. What automated tests I added (or what prevented me from doing so)

    - Updated existing tests

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] ~WordPress.com sites and self-hosted Jetpack sites.~
- [ ] ~Portrait and landscape orientations.~
- [ ] ~Light and dark modes.~
- [ ] ~Fonts: Larger, smaller and bold text.~
- [ ] ~High contrast.~
- [x] Talkback.
- [ ] ~Languages with large words or with letters/accents not frequently used in English.~
- [ ] ~Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)~
- [ ] ~Large and small screen sizes. (Tablet and smaller phones)~
- [ ] ~Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><p>This PR adds a translatable entry which references another string resource; this usually causes issues with translations.<br />
Please make sure to set the <code>translatable="false"</code> attribute.<br />
File <code>WordPress/src/main/res/values/strings.xml</code>:</p>
<div class="language-diff highlighter-rouge"><pre class="highlight"><code><span class="gi">+        &lt;item&gt;@string/site_subscriptions&lt;/item&gt;
</span></code></pre></div></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1322,11 +1322,9 @@
     <string name="stats_view_comments" translatable="false">@string/comments</string>
     <string name="stats_view_search_terms">Search Terms</string>
     <string name="stats_view_publicize">Jetpack Social Connections</string>
-    <string name="stats_view_followers">Followers</string>
-    <string name="stats_view_follower_totals">Follower Totals</string>
+    <string name="stats_view_subscriber_totals">Subscriber Totals</string>
     <string name="stats_view_total_likes">Total Likes</string>
     <string name="stats_view_total_comments">Total Comments</string>
-    <string name="stats_view_total_followers">Total Followers</string>
```
Comment (irfano) [inline WordPress/src/main/res/values/strings.xml:1328]: Thank you for also removing these unused strings. Probably, I forgot to do it while removing the "Followers Total" card.

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1607,14 +1605,14 @@
     <string name="comments_on_my_site">Comments on my site</string>
     <string name="likes_on_my_comments">Likes on my comments</string>
     <string name="likes_on_my_posts">Likes on my posts</string>
-    <string name="site_follows">Site follows</string>
+    <string name="site_subscribers">Site subscribers</string>
```
Comment (irfano) [inline WordPress/src/main/res/values/strings.xml]: "Site subscriptions" would fit better here. That's how iOS did it.

File: WordPress/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -2649,7 +2647,7 @@
     <string name="follower_subscribed_since">Since %1$s</string>
     <string name="person_remove_confirmation_title">Remove %1$s</string>
     <string name="user_remove_confirmation_message">If you remove %1$s, that user will no longer be able to access this site, but any content that was created by %1$s will remain on the site.\n\nWould you still like to remove this user?</string>
-    <string name="follower_remove_confirmation_message">If removed, this follower will stop receiving notifications about this site, unless they re-follow.\n\nWould you still like to remove this follower?</string>
+    <string name="subscriber_remove_confirmation_message">If removed, this subscriber will stop receiving notifications about this site, unless they re-follow.\n\nWould you still like to remove this subscriber?</string>
```
Comment (irfano) [inline WordPress/src/main/res/values/strings.xml]: "..., unless they re-follow..." → "..., unless they re-subscribe..."

**Review by irfano — CHANGES_REQUESTED** [review]
Good work! I've requested a few changes to the strings. Everything looks good except for those changes.

Also, what do you think about renaming the enum names (`PeopleListFilter`, `TabPosition`, `ALREADY_FOLLOWING`)? I'll let you decide. Feel free to update them if they seem appropriate and don't necessitate extensive changes.

File: WordPress/src/main/res/values/strings.xml (hunk 2 above)
Comment (aditi-bhatia) [reply WordPress/src/main/res/values/strings.xml]: Thanks! You're right, that does sound better 👍

File: WordPress/src/main/res/values/strings.xml (hunk 3 above)
Comment (aditi-bhatia) [reply WordPress/src/main/res/values/strings.xml]: Good catch, thank you!

**Comment by aditi-bhatia** [comment]
> Also, what do you think about renaming the enum names (PeopleListFilter, TabPosition, ALREADY_FOLLOWING)? I'll let you decide. Feel free to update them if they seem appropriate and don't necessitate extensive changes.

I considered it but felt it might make things more confusing 😅  For example, `StatsViewType` has

-  `FOLLOWERS`, 
- `SUBSCRIBERS`, and 
- `TOTAL_FOLLOWERS` currently. 

Similarly, `StatsType` (within `StatsStore.kt`) currently has 

- `FOLLOWERS`, 
- `FOLLOWER_TOTALS`, 
- `TOTAL_FOLLOWERS` 
 and 
- `TOTAL_SUBSCRIBERS`, 
- `SUBSCRIBERS_CHART`, 
-  `SUBSCRIBERS`

Renaming all of the `follower` occurrences above to `subscribers` would result in duplicate names unless we came up with a new naming convention, so I decided to leave them alone for now since I believe we might be removing some of that code in the future. What do you think?

**Review by irfano — APPROVED** [review]
I agree that renaming stats would be confusing. That's why I didn't include them in the example classes in my comment. 😄 
Just consider `PeopleListFilter`, `TabPosition`, `ALREADY_FOLLOWING`, which are not related to Stats. I'm approving, feel free to merge.

**Comment by aditi-bhatia** [comment]
That sounds good to me, I went ahead and made some changes to the enum namings 👍

**Pull request merged**